### PR TITLE
Path parm attribute

### PR DESCRIPTION
**Pull request by puggan**
Add a new Attribute, that allow you to define path-parameters on the method.

### CODE REVIEW

File: src/Attributes/Route.php [hunk 1]
```diff
@@ -64,6 +65,11 @@ public function getGetParams(): array
         return $this->getParams;
     }
 
+    /**
+     * @param Parameter[][] $getParams
+     * @param ReflectionAttribute[] $pathParameters
+     * @return void
+     */
```
Comment (uderline) [inline src/Attributes/Route.php:72]: Not sure this is necessary. We can simply put the following if you think it's relevant to the readability of the code
```
/**
 * @param Parameter[][] $getParams
 */
```

Comment (puggan) [reply src/Attributes/Route.php:72]: Oh, that was automaticly added when I was testing, the new parameter is removed, but I didn't remove the autogenerated phpdoc.

**Comment by puggan** [comment]
If we go with #15, the parameters added here needs to be faltten to.

https://github.com/uderline/openapi-php-attributes/blob/99382fc1fea7190adbbd2a657c0c3090376c070b/src/GeneratorHttp.php#L49-L54

File: src/PathMethodBuilder.php [hunk 2]
```diff
@@ -29,7 +30,7 @@ class PathMethodBuilder
      * Set the current route with GET parameters
      *
      * @param Route $route
-     * @param array $params
+     * @param Parameter[][] $params
```
Comment (uderline) [inline src/PathMethodBuilder.php:33]: And here ;)

**Comment by uderline** [comment]
After testing, I found an issue on the `IDParam`'s constructor:
```php
public function __construct(
        private ?string $description = null,
        private string $in = 'path',
        private ?bool $required = null,
    ) {
```

Which must become:
```php
public function __construct(
        protected ?string $description = null,
        protected string $in = 'path',
        protected ?bool $required = null,
    ) {
```

Could you change `SimpleController` to this (line 34) please so we can have a test ?
```php
#[
        GET("/path/{id}/{otherParameter}", ["Dummy"], "Dummy path"),
        PathParameter("otherParameter", description: "Parameter which is not used as an argument in this method"),
        Property(Type::STRING, "prop1"),
        Response(200),
    ]
```

**Comment by puggan** [comment]
Now when `Parameter::$description` is protected (not private), it's accessable from IDParam, so there is no need to duplicate it by adding a variable also at `IDParam::$description`, so just removed the private from IDParam.
(Also give you permission to my repo, so you easy can just pres edit in githubs web-gui, and do small changes without the need of asking me).

**Pull request merged**